src/main/webapp/WEB-INF/pages/checkout.jsp

```diff
@@ -0,0 +1,106 @@
+    <form method="post" action="${pageContext.request.contextPath}/checkout">
+        <table>
+            <thead>
+            <tr>
+                <td>Image</td>
+                <td>
+                    Description
+                </td>
+                <td>Quantity</td>
+                <td class="price">
+                    Price
+                </td>
+            </tr>
+            </thead>
+            <c:forEach var="item" items="${order.items}" varStatus="status">
+                <tr>
+                    <td>
+                        <img class="product-tile" src=${item.product.imageUrl}>
+                    </td>
+                    <td>
+                        <a href="${pageContext.request.contextPath}/products/${item.product.id}">
+                                ${item.product.description}
+                        </a>
+                    </td>
+                    <td>
+                            ${item.quantity}
+                    </td>
+                    <td class="price">
+                        <a href="#popup${item.product.id}">
+                            <fmt:formatNumber value="${item.product.price}" type="currency"
+                                              currencySymbol="${item.product.currency.symbol}"/>
+                        </a>
+                        <div id="popup${item.product.id}" class="overlay">
+                            <tags:priceHistory product="${item.product}"/>
+                        </div>
+                    </td>
+                </tr>
+            </c:forEach>
+                <tr>
+                    <td></td>
+                    <td></td>
+                    <td>Total Quantity: ${order.totalQuantity}</td>
+                    <td class="price"> Subtotal: <tags:price order="${order}" paramName="subTotal"/>
+                    </td>
+                </tr>
+                <tr>
+                    <td></td>
+                    <td></td>
+                    <td></td>
+                    <td class="price">Delivery cost: <tags:price order="${order}" paramName="deliveryCost"/>
+                    </td>
+                </tr>
+                <tr>
+                    <td></td>
+                    <td></td>
+                    <td></td>
+                    <td class="price">Total cost: <tags:price order="${order}" paramName="totalCost"/>
+                    </td>
+                </tr>
+        </table>
+        <h2>Your Details</h2>
+            <table>
+                <tags:setParameter paramName="firstName" order="${order}" label="First name" errors="${errors}" type="text"/>
+                <tags:setParameter paramName="lastName" order="${order}" label="Last name" errors="${errors}" type="text"/>
+                <tags:setParameter paramName="phoneNumber" order="${order}" label="Phone" errors="${errors}" type="text"/>
+                <tags:setParameter paramName="deliveryDate" order="${order}" label="Delivery date" errors="${errors}" type="date"/>
```

Comment:
Please add some hint or anything so that user will understand how he can set delivery date. Atm it's absolutely unclear

Reply:
And `setParameter` is quite strange name for a tag - it's confusing for this tag name to contain action

Reply:
I added input type to this tag
![image](https://user-images.githubusercontent.com/63584690/109515010-a9204400-7ab7-11eb-9a3d-e5f564bf9008.png)
So it looks like this
![image](https://user-images.githubusercontent.com/63584690/109515071-bb01e700-7ab7-11eb-9a7b-2b94eac8c6d6.png)